### PR TITLE
Hunter gear teleport ability

### PR DESCRIPTION
**Pull request by flappybatpal**
[FEATURE]

## About the PR 
- New spear sprite with real inhands!
- Adds an ability to hunters to teleport their gear to their location on a 40 second cooldown.
- Starts hunter spear and rifle in their backpack.

image

https://user-images.githubusercontent.com/70909958/150685198-887a0120-2102-43de-80d7-367dabb2f7c8.mp4

## Why's this needed? 
- Spear is more legible and will be easier to click on.
- Hunters are gear based antags with limited other abilities (door opening, better melee) who lose a lot of their power if they drop their gear. Observing hunters reveals this is pretty common!
- Spear and rifle would not spawn if you were missing an arm, which is not great! As the hunter will be equipped with a backpack on transform there should not be any way for them to not receive or instantly lose their gear.

## Changelog

```changelog
(u)Flappybat and Flaborized
(*)Added a Recover Gear ability to Hunters. Use it to teleport your rifle, spear or cloak back to your position if you lose them. Animations by Flaborized.
(*)New Hunting spear sprite and inhands by Flaborized!
```

### CODE REVIEW

File: code/datums/abilities/misc/hunter_summongear.dm [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		var/mob/living/M = holder.owner
+
+		if (!M || !ishuman(M))
+			return 1
+
+		actions.start(new/datum/action/bar/private/icon/hunter_summongear(src), M)
+		return 0
+
+/datum/action/bar/private/icon/hunter_summongear
+	duration = 50
```
Comment (pali6) [inline code/datums/abilities/misc/hunter_summongear.dm:26]: Should probably be `5 SECONDS`

File: code/datums/abilities/misc/hunter_summongear.dm [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	onEnd()
+		..()
+
+		var/mob/living/M = owner
+		var/gun_found = FALSE
+		var/spear_found = FALSE
+		var/cloak_found = FALSE
+		for(var/HG in by_cat[TR_CAT_HUNTER_GEAR])
+			if (istype(HG, /obj/item/gun/energy/plasma_gun/hunter))
```
Comment (pali6) [inline code/datums/abilities/misc/hunter_summongear.dm:60]: In this part of the code `M.mind && M.mind.key == ...` could be replaced with `M.mind?.key = ...`.

Also the continue statements don't seem to be necessary as they are always at the end of each branch.

Comment (pali6) [inline code/datums/abilities/misc/hunter_summongear.dm:60]: In this part of the code `M.mind && M.mind.key == ...` could be replaced with `M.mind?.key = ...`.

Also the continue statements don't seem to be necessary as they are always at the end of each branch.

Comment (pali6) [inline code/datums/abilities/misc/hunter_summongear.dm:60]: In this part of the code `M.mind && M.mind.key == ...` could be replaced with `M.mind?.key = ...`.

Also the continue statements don't seem to be necessary as they are always at the end of each branch.

File: code/datums/abilities/misc/hunter_summongear.dm [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+				if (M.mind && M.mind.key == HC.hunter_key)
+					if (get_turf(M.loc) == get_turf(HC.loc))
+						cloak_found = TRUE
+						continue
+					else
+						HC.send_to_target_mob(M)
+						cloak_found = TRUE
+						continue
+		if (!gun_found)
+			boutput(M, __red("Your plasma gun is lost or destroyed!"))
```
Comment (pali6) [inline code/datums/abilities/misc/hunter_summongear.dm:91]: I think the __red stuff is pretty deprecated but not sure. I'd just manually enter the span with the alert class.

File: code/obj/item/gun/energy.dm [hunk 4]
```diff
@@ -1027,12 +1027,42 @@
...
 			src.AddComponent(/datum/component/self_destruct, M)
+			src.hunter_key = M.mind.key
+			START_TRACKING_CAT(TR_CAT_HUNTER_GEAR)
+			flick("[src.base_item_state]-tele", src)
+
+	disposing()
+		. = ..()
+		STOP_TRACKING_CAT(TR_CAT_HUNTER_GEAR)
+
+	proc/send_to_target_mob(var/mob/living/M)
```
Comment (pali6) [inline code/obj/item/gun/energy.dm:1045]: The copypasted proc is not very neat. Ideally I think this all would be a component. But if you don't feel like making it a component probably make the proc a global one instead.

**Review by pali6 — CHANGES_REQUESTED** [review]

**Review by TTerc — APPROVED** [review]

**Pull request merged**